unit testing code for l2_interfaces module added

kannankvs opened the pull request:
##### SUMMARY
l2_interfaces module Unit Test Code Added

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
l2_interfaces module Unit Testing

kannankvs commented:
'replaced' and 'overridden' features are getting modified in l2_interfaces module by developers in parallel. This UT PR can be merged once if the PR for l2_interfaces module for 'replaced' and 'overridden' support is merged. If this UT requires rework after merging the l2_interfaces changes, it will be taken up after that gets merged.

kerry-meyer commented on plugins/module_utils/network/sonic/argspec/l2_interfaces/l2_interfaces.py:
Please delete this file from the change set. Let's wait until the real "replaced" and "overridden" support is added to do these changes. (This comment applies to any other UT modules where the module and argspec are being updated to add these states before the underlying code support has been provided.)

kerry-meyer commented on plugins/modules/sonic_l2_interfaces.py:
Please delete this file from the change set. Let's wait until the real "replaced" and "overridden" support is added to do these changes. (This comment applies to any other UT modules where the module and argspec are being updated to add these states before the underlying code support has been provided.)

kerry-meyer replied:
Alternatively, this PR will have to wait until after PR https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/221 is merged.

kerry-meyer replied:
Thanks for making the additional changes.

kerry-meyer reviewed (changes requested):
Please delete the argspec and module file from the change set. Let's wait until the real "replaced" and "overridden" support is added to do these changes. (This comment applies to any other UT modules where the module and argspec are being updated to add these states before the underlying code support has been provided.)

kannankvs commented:
Changes related to "replaced" and "overridden" in plugins directory are not required and hence removed them now. They should be synced from PR221

kannankvs commented:
@kerry-meyer : I removed code for testing replaced and overridden. When I remove UT for replaced and overridden, the coverage percentage decreases below 70. Hence, I added few new combinations and improved the coverage above 70% now. Please review and merge.
